### PR TITLE
fix upgrade fail check condition to be compatible with 2.2.0

### PR DESCRIPTION
**Pull request by zhiweiyin318**

### CODE REVIEW

File: pkg/jobs/utils/helpers.go [hunk 1]
```diff
@@ -378,7 +378,12 @@ func NeedToUpgrade(curator clustercuratorv1.ClusterCurator) (bool, error) {
 	if strings.Contains(jobCondtion.Message, "Failed") {
 		klog.V(2).Info(fmt.Sprintf("Previous curator %q is failed, %q", curator.Name, jobCondtion.Message))
 		if strings.Contains(jobCondtion.Message, GetCurrentVersionInfo(&curator)) {
-			// last job failed and desired version is unchange, do nothing
+			// last job failed and desired version is unchanged, do nothing
```
Comment (skeeey) [inline pkg/jobs/utils/helpers.go:381]: we may also need add log info here

File: pkg/jobs/utils/helpers.go [hunk 2]
```diff
@@ -378,7 +378,12 @@ func NeedToUpgrade(curator clustercuratorv1.ClusterCurator) (bool, error) {
...
+			return false, nil
+		}
+
+		// To be compatible with 2.2.0 version
+		if strings.Contains(jobCondtion.Message, fmt.Sprintf("Version (%s)", curator.Spec.Upgrade.DesiredUpdate)) {
```
Comment (skeeey) [inline pkg/jobs/utils/helpers.go:386]: we may also need add log info here

File: pkg/jobs/utils/helpers.go (hunk 1 above)
Comment (zhiweiyin318) [reply pkg/jobs/utils/helpers.go:381]: done

File: pkg/jobs/utils/helpers.go (hunk 2 above)
Comment (zhiweiyin318) [reply pkg/jobs/utils/helpers.go:387]: done

**Comment by skeeey** [comment]
LGTM

**Comment by elgnay** [comment]
LGTM

@xiangjingli You may also want to take a look at this PR. It will fix the issue: https://issues.redhat.com/browse/ACM-4196

**Pull request merged**